### PR TITLE
B/startup

### PR DESCRIPTION
**Pull request by worldofjoni**
# Pull Request for initial Implementation

## Backend
Make sure, you have done de following before merging this pull request:
- [x] Your code is **documented** using doc comments.
- [x] You added appropriate **logging** in your code.
- [x] You added **unit tests** to all your implemented functions.
- [x] You ran `cargo format` to format your code.
- [x] You ran `cargo clippy` to check for improvements.
- [x] You added the **class descriptions** of the classes you implemented in the **latex document** on overleaf.
      Make sure to mark any changes you made to the design documents specification.

### CODE REVIEW

File: backend/.env.example [hunk 1]
```diff
@@ -1,11 +1,33 @@
 # copy this file to .env and set your own values
+# optional enviroenmant variabls (when default values are available) are commented out
 
-# connection to database
+# --- connection to database ---
 DATABASE_URL=postgres://postgres_user@localhost/mensa_app
 
-#for sending mails over smtp
+# --- ails over smtp ---
```
Comment (uglrl) [inline backend/.env.example:7]: snails?

File: backend/src/layer/data/flickr_api/api_request.rs [hunk 2]
```diff
@@ -120,7 +120,7 @@ mod test {
 
     fn get_api_key() -> String {
         dotenv().ok();
-        env::var("FLICKR_PUBLIC_KEY").expect("FLICKR_PUBLIC_KEY should be set in the .env!")
+        env::var("FLICKR_API_KEY").expect("FLICKR_API_KEY should be set in the .env!")
```
Comment (uglrl) [inline backend/src/layer/data/flickr_api/api_request.rs:123]: You might need to change this in the tests as well

File: backend/src/layer/data/flickr_api/flickr_api_handler.rs [hunk 3]
```diff
@@ -22,18 +24,19 @@ lazy_static! {
         Regex::new(r"(https://flic.kr/p/)([\d\w]+)").expect("regex creation failed");
 }
 
-impl FlickrApiHandler {
+impl FlickeApiHandler {
```
Comment (uglrl) [inline backend/src/layer/data/flickr_api/flickr_api_handler.rs:27]: Typo

File: backend/src/layer/data/swka_parser/swka_parse_manager.rs [hunk 4]
```diff
@@ -27,7 +27,7 @@ impl SwKaParseManager {
     /// Method for creating a [`SwKaParseManager`] instance.
```
Comment (uglrl) [inline backend/src/layer/data/swka_parser/swka_parse_manager.rs:27]: Update docs. The name is outdated

**Review by uglrl — CHANGES_REQUESTED** [review]

File: backend/src/startup/config.rs [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        Ok(info)
+    }
+
+    /// Reads the schedules for regular events from environment variables.
+    /// # Errors
+    /// when the environment variables are not set and no default is provided internally.
+    pub fn read_schedule_info(&self) -> Result<ScheduleInfo> {
+        let info = ScheduleInfo {
+            full_parse_schedule: env::var("FULL_PARSE_SCHEDULE")
+                .unwrap_or_else(|_| "0 0 2 * * *".into()),
```
Comment (uglrl) [inline backend/src/startup/config.rs:90]: Maybe refactor the cron times as constants

File: backend/src/startup/config.rs [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        env::args().any(|arg| HELP.contains(&arg.as_str()))
+    }
+
+    /// Reads the logging configuration from environment variables.
+    /// # Errors
+    /// when the environment variables are not set and no default is provided internally.
+    pub fn read_log_info(&self) -> Result<LogInfo> {
+        let info = LogInfo {
+            log_config: read_var("LOG_CONFIG")
+                .unwrap_or_else(|_| "warn,mensa_app_backend=info".into()),
```
Comment (uglrl) [inline backend/src/startup/config.rs:51]: Make this a constant variable

File: backend/src/startup/config.rs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    /// Reads the config for the homepage of the "Studierendenwerk Karlsruhe" (Sw Ka) and its canteens from environment variables.
+    /// # Errors
+    /// when the environment variables are not set and no default is provided internally.
+    pub fn read_swka_info(&self) -> Result<SwKaInfo> {
+        let timeout = env::var("CLIENT_TIMEOUT")
+            .ok()
+            .and_then(|s| s.parse().ok())
+            .unwrap_or(6000);
```
Comment (uglrl) [inline backend/src/startup/config.rs:116]: 6000 -> constant

File: backend/src/startup/config.rs [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+            .split(',')
+            .map(str::trim)
+            .map(String::from)
+            .collect();
+
+        let info = SwKaInfo {
+            base_url: read_var("MENSA_BASE_URL").unwrap_or_else(|_| DEFAULT_BASE_URL.into()),
+            client_timeout: timeout,
+            client_user_agent: env::var("USER_AGENT")
+                .unwrap_or_else(|_| String::from("MensaKa 0.1")),
```
Comment (uglrl) [inline backend/src/startup/config.rs:130]: see above comments

File: backend/src/startup/config.rs [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+    /// Reads the config for the graphql web server from environment variables.
+    /// # Errors
+    /// when the environment variables are not set and no default is provided internally.
+    pub fn read_graphql_info(&self) -> Result<GraphQLServerInfo> {
+        let info = GraphQLServerInfo {
+            port: env::var("HTTP_PORT")
+                .ok()
+                .and_then(|p| p.parse().ok())
+                .unwrap_or(80),
```
Comment (uglrl) [inline backend/src/startup/config.rs:149]: S. a.

File: backend/src/layer/trigger/scheduling/scheduler.rs [hunk 10]
```diff
@@ -177,14 +177,14 @@ mod tests {
...
-    use tracing::Level;
+    // use tracing::Level;
     #[tokio::test]
     async fn test_scheduling() {
-        let subscriber = tracing_subscriber::FmtSubscriber::builder()
-            .with_max_level(Level::TRACE)
-            .finish();
-        tracing::subscriber::set_global_default(subscriber)
-            .expect("Setting default subscriber failed");
+        // let subscriber = tracing_subscriber::FmtSubscriber::builder()
```
Comment (uglrl) [inline backend/src/layer/trigger/scheduling/scheduler.rs:183]: Why did you comment this?

**Review by uglrl — CHANGES_REQUESTED** [review]

File: backend/src/layer/data/flickr_api/api_request.rs (hunk 2 above)
Comment (worldofjoni) [reply backend/src/layer/data/flickr_api/api_request.rs:123]: do you mean the test in github?

File: backend/src/layer/data/swka_parser/swka_parse_manager.rs (hunk 4 above)
Comment (worldofjoni) [reply backend/src/layer/data/swka_parser/swka_parse_manager.rs:27]: no?

File: backend/src/layer/trigger/scheduling/scheduler.rs (hunk 10 above)
Comment (worldofjoni) [reply backend/src/layer/trigger/scheduling/scheduler.rs:183]: these were the spam comments you complaint about

File: backend/src/layer/data/flickr_api/api_request.rs (hunk 2 above)
Comment (uglrl) [reply backend/src/layer/data/flickr_api/api_request.rs:123]: No the ones from @Whatsuup

File: backend/src/layer/trigger/scheduling/scheduler.rs (hunk 10 above)
Comment (uglrl) [reply backend/src/layer/trigger/scheduling/scheduler.rs:183]: Spam comments?

File: backend/src/layer/data/flickr_api/api_request.rs (hunk 2 above)
Comment (worldofjoni) [reply backend/src/layer/data/flickr_api/api_request.rs:123]: aren't these the tests?

File: backend/src/layer/trigger/scheduling/scheduler.rs (hunk 10 above)
Comment (worldofjoni) [reply backend/src/layer/trigger/scheduling/scheduler.rs:183]: *log messegas

Comment (worldofjoni) [reply backend/src/layer/trigger/scheduling/scheduler.rs:183]: i can remove these completely

Comment (uglrl) [reply backend/src/layer/trigger/scheduling/scheduler.rs:183]: Yes please

**Review by uglrl — APPROVED** [review]

**Pull request merged**